cfbs/commands.py

```diff
@@ -1023,9 +1023,10 @@ def set_input_command(name, infile):
         return 1
     log.debug("Input data for module '%s': %s" % (name, pretty(data)))
 
-    def _compare_dict(a, b, ignore=set()):
+    def _compare_dict(a, b, ignore=None):
+        assert ignore is None or isinstance(ignore, set)
         assert isinstance(a, dict) and isinstance(b, dict)
-        if set(a.keys()) != set(b.keys()) - ignore:
+        if set(a.keys()) != set(b.keys()) - (set() if ignore is None else ignore):
```

Comment:
The common trick is to do `ignore = ignore or set()` (or `ignore = ignore if ignore is not None else set()`) at the beginning of the function.